docker/Dockerfile-mariner

```diff
@@ -0,0 +1,39 @@
+# Adapted from https://github.com/dotnet/dotnet-docker/blob/3c17243313fe40456e930b64e44d45e6829e19f3/src/runtime-deps/6.0/cbl-mariner2.0-distroless/amd64/Dockerfile
+# Copyright (c) 2016 .NET Foundation
+# License: MIT (https://github.com/dotnet/dotnet-docker/blob/3c17243313fe40456e930b64e44d45e6829e19f3/LICENSE)
+
+# Note: current directory must be ./dist
+
+### Builder image ###
+ARG MARINER_VERSION=2.0
+FROM mcr.microsoft.com/cbl-mariner/base/core:${MARINER_VERSION} AS builder
+
+# Install dependencies into a staging location
+ARG MARINER_VERSION
+RUN mkdir /staging \
+    && tdnf install -y --releasever="$MARINER_VERSION" --installroot /staging \
+        prebuilt-ca-certificates \
+        tzdata \
```

Comment:
tzdata is included because we have it in the distroless images too. However, it's unclear if this is really needed... It's a pretty small package, just a few KBs, but we may not really need tzdata for Dapr.